app/collection.go

```diff
@@ -4,37 +4,36 @@
 package app
 
 import (
+	"fmt"
 	"net/http"
 
 	"github.com/mattermost/mattermost-server/v6/model"
 	"github.com/mattermost/mattermost-server/v6/utils"
 )
 
 func (a *App) registerCollectionAndTopic(pluginID, collectionType, topicType string) error {
-	if !a.Config().FeatureFlags.ThreadsEverywhere {
-		return model.NewAppError("registerCollectionAndTopic", "app.collection.feature_flag.app_error", nil, "", http.StatusBadRequest)
-	}
 	// we have a race condition due to multiple plugins calling this method
 	a.ch.collectionAndTopicTypesMut.Lock()
 	defer a.ch.collectionAndTopicTypesMut.Unlock()
 
 	// check if collectionType was already registered by other plugin
-	for plugID, colTypes := range a.ch.collectionTypes {
-		if utils.StringInSlice(collectionType, colTypes) && plugID != pluginID {
+	for existingPluginID, existingCollectionTypes := range a.ch.collectionTypes {
+		if existingPluginID != pluginID && utils.StringInSlice(collectionType, existingCollectionTypes) {
 			return model.NewAppError("registerCollectionAndTopic", "app.collection.add_collection.exists.app_error", nil, "", http.StatusBadRequest)
 		}
 	}
 
 	// check if topicType was already registered to other collection
-	for collectionTypeForTopic, topicTypes := range a.ch.topicTypes {
-		if utils.StringInSlice(topicType, topicTypes) && collectionTypeForTopic != collectionType {
+	for existingCollectionType, existingTopicTypes := range a.ch.topicTypes {
+		if existingCollectionType != collectionType && utils.StringInSlice(topicType, existingTopicTypes) {
 			return model.NewAppError("registerCollectionAndTopic", "app.collection.add_topic.exists.app_error", nil, "", http.StatusBadRequest)
 		}
 	}
 
 	a.ch.collectionTypes[pluginID] = appendIfUnique(a.ch.collectionTypes[pluginID], collectionType)
 	a.ch.topicTypes[collectionType] = appendIfUnique(a.ch.topicTypes[collectionType], topicType)
 
+	a.ch.srv.Log().Info(fmt.Sprintf("%s registered collection type %s and topic type %s", pluginID, collectionType, topicType))
```

Comment:
Can we pass the fields as structured arguments?
```suggestion
	a.ch.srv.Log().Info("registered collection and topic type", mlog.String("plugin_id", pluginID), mlog.String("collection_type", collectionType), mlog.String("topic_type", topicType))
```